znft/ethereum_wallet.go

```diff
@@ -0,0 +1,93 @@
+package znft
+
+import (
+	"context"
+	"math/big"
+	"path"
+	"time"
+
+	"github.com/ethereum/go-ethereum/accounts"
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/accounts/keystore"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/ethclient"
+	"github.com/ethereum/go-ethereum/params"
+	"github.com/pkg/errors"
+)
+
+func (conf *Configuration) CreateEthClient() (*ethclient.Client, error) {
```

Comment:
We'd better not creating methods on configuration, instead, we should pass the config as param, so as all the other following methods. Also, we can pass necessary params only, so for the `CreateEthClient()`, we will only need to pass the `EthereumNodeURL`. For the `createSignedTransactionFromKeyStore()` we will only need to pass the password and value, etc.

Reply:
Updated in be39946fd2c31c03f4a392442e4151b601b11b16